packages/flutter/lib/src/material/overscroll_painter.dart

```diff
@@ -0,0 +1,133 @@
+
+typedef Color GetOverscrollIndicatorColor();
+
+class OverscrollPainter extends ScrollableListPainter {
+  OverscrollPainter({ GetOverscrollIndicatorColor getIndicatorColor }) {
+    this.getIndicatorColor = getIndicatorColor ?? _defaultIndicatorColor;
+  }
+
+  GetOverscrollIndicatorColor getIndicatorColor;
+  bool _indicatorActive = false;
+  AnimationController _indicatorLength;
+  Timer _indicatorTimer;
+
+  Color _defaultIndicatorColor() => const Color(0xFF00FF00);
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (_indicatorLength == null || (scrollOffset >= _minScrollOffset && scrollOffset <= _maxScrollOffset))
+      return;
+
+    final double rectBias = _indicatorLength.value / 2.0;
+    final double arcBias = _indicatorLength.value;
+    final Rect viewportRect = offset & viewportSize;
+
+    final Path path = new Path();
+    switch(scrollDirection) {
+      case Axis.vertical:
+        final double width = viewportRect.width;
+        if (scrollOffset < _minScrollOffset) {
+          path.moveTo(viewportRect.left, viewportRect.top);
+          path.relativeLineTo(width, 0.0);
+          path.relativeLineTo(0.0, rectBias);
+          path.relativeQuadraticBezierTo(width / -2.0, arcBias, -width, 0.0);
+        } else {
+          path.moveTo(viewportRect.left, viewportRect.bottom);
+          path.relativeLineTo(width, 0.0);
+          path.relativeLineTo(0.0, -rectBias);
+          path.relativeQuadraticBezierTo(width / -2.0, -arcBias, -width, 0.0);
+        }
+        break;
+      case Axis.horizontal:
+        final double height = viewportRect.height;
+        if (scrollOffset < _minScrollOffset) {
+          path.moveTo(viewportRect.left, viewportRect.top);
+          path.relativeLineTo(0.0, height);
+          path.relativeLineTo(rectBias, 0.0);
+          path.relativeQuadraticBezierTo(arcBias, height / -2.0, 0.0, -height);
+        } else {
+          path.moveTo(viewportRect.right, viewportRect.top);
+          path.relativeLineTo(0.0, height);
+          path.relativeLineTo(-rectBias, 0.0);
+          path.relativeQuadraticBezierTo(-arcBias, height / -2.0, 0.0, -height);
+        }
+        break;
+    }
+    path.close();
+
+    final double t = (_indicatorLength.value - _kMinIndicatorLength) / (_kMaxIndicatorLength - _kMinIndicatorLength);
+    final double opacity = new Tween(begin: 0.0, end: 0.3).lerp(Curves.easeIn.transform(t));
+    final Paint paint = new Paint()..color = getIndicatorColor().withOpacity(opacity);
+    context.canvas.drawPath(path, paint);
+  }
+
+  void _hide() {
+    _indicatorTimer?.cancel();
+    _indicatorTimer = null;
+    _indicatorActive = false;
+    _indicatorLength?.reverse();
+  }
+
+  double get _minScrollOffset => 0.0;
+
+  double get _maxScrollOffset {
+    switch(scrollDirection) {
+      case Axis.vertical:
+        return contentExtent - viewportSize.height;
+      case Axis.horizontal:
+        return contentExtent - viewportSize.width;
+    }
+  }
```

Comment:
These aren't necessarily true.  Maybe we should have a reference to the scroll behavior so we can read the proper values?  Specifically, there are many cases where the min scroll offset isn't zero.